### PR TITLE
[RLlib] Add testing framework_iterator.

### PR DESCRIPTION
**Pull request by sven1977**
All RLlib test cases should be framework-agnostic (if applicable, meaning the test is affecting framework-dependent components). This PR introduces a framework_iterator for creating simple for loops in test cases that automatically take care of eager-context creation, tf-test-session creation and useful automatic assertions.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/agents/ddpg/tests/test_ddpg.py [hunk 1]
```diff
@@ -15,11 +15,7 @@ def test_ddpg_compilation(self):
         config["num_workers"] = 0  # Run locally.
 
         # Test against all frameworks.
-        for fw in ["tf", "eager", "torch"]:
-            if fw != "tf":
-                continue
-            config["eager"] = True if fw == "eager" else False
-            config["use_pytorch"] = True if fw == "torch" else False
+        for _ in framework_iterator(config, "tf"):
```
Comment (ericl) [inline rllib/agents/ddpg/tests/test_ddpg.py:18]: Shouldn't the second arg be a tuple?

Comment (sven1977) [reply rllib/agents/ddpg/tests/test_ddpg.py:18]: It can handle both. `str` or `Tuple[str]`.

**Comment by sven1977** [comment]
@ericl Please merge. Only the `local_mode_manager` ray-core error occurring. All others tests are ok.

**Comment by sven1977** [comment]
@ekl

**Review by ericl — APPROVED** [review]

**Pull request merged**